app/controllers/sessions_controller.rb

```diff
@@ -1,4 +1,15 @@
 class SessionsController < ApplicationController
   def new
   end
+
+  def create
+    user = User.find_by(username: params[:session][:username])
+    if user && user.authenticate(params[:session][:password])
+      session[:user_id] = user.id
+      redirect_to dashboard_path
+    else
+      flash.now[:error] = "Username and/or Password is invalid. Try again."
+      render :new
+    end
+  end
```

Comment:
I see hound's point that this is a little long, but I am not really sure what you would refactor. I think it is fine